File: db/mysql/driver.go
package mysql

import (
	"context"

	"github.com/lazyxu/kfs/dao"
	"github.com/lazyxu/kfs/db/dbBase"
)

func (db *DB) InsertDriver(ctx context.Context, driverName string, description string) (exist bool, err error) {
	return dbBase.InsertDriver(ctx, db.db, db, driverName, description)
}

func (db *DB) InsertDriverBaiduPhoto(ctx context.Context, driverName string, description string, accessToken string, refreshToken string) (exist bool, err error) {
	return dbBase.InsertDriverBaiduPhoto(ctx, db.db, db, driverName, description, accessToken, refreshToken)
}

func (db *DB) InsertDriverLocalFile(ctx context.Context, driverName string, description string, deviceId uint64, srcPath string, ignores string, encoder string) (exist bool, err error) {
	return dbBase.InsertDriverLocalFile(ctx, db.db, db, driverName, description, deviceId, srcPath, ignores, encoder)
}

func (db *DB) UpdateDriverSync(ctx context.Context, driverId uint64, sync bool, h int64, m int64) error {
	return dbBase.UpdateDriverSync(ctx, db.db, driverId, sync, h, m)
}

func (db *DB) UpdateDriverLocalFile(ctx context.Context, driverId uint64, srcPath, ignores, encoder string) error {
	return dbBase.UpdateDriverLocalFile(ctx, db.db, driverId, srcPath, ignores, encoder)
}

func (db *DB) ResetDriver(ctx context.Context, driverId uint64) error {
	return dbBase.ResetDriver(ctx, db.db, driverId)
}

func (db *DB) DeleteDriver(ctx context.Context, driverId uint64) error {
	return dbBase.DeleteDriver(ctx, db.db, driverId)
}

func (db *DB) ListDriver(ctx context.Context) (drivers []dao.Driver, err error) {
	return dbBase.ListDriver(ctx, db.db)
}

func (db *DB) GetDriver(ctx context.Context, driverId uint64) (driver dao.Driver, err error) {
	return dbBase.GetDriver(ctx, db.db, driverId)
}

func (db *DB) GetDriverToken(ctx context.Context, driverId uint64) (driver dao.Driver, err error) {
	return dbBase.GetDriverToken(ctx, db.db, driverId)
}

func (db *DB) GetDriverSync(ctx context.Context, driverId uint64) (driver dao.Driver, err error) {
	return dbBase.GetDriverSync(ctx, db.db, driverId)
}

func (db *DB) ListCloudDriverSync(ctx context.Context) (drivers []dao.Driver, err error) {
	return dbBase.ListCloudDriverSync(ctx, db.db)
}

func (db *DB) ListLocalFileDriver(ctx context.Context, deviceId uint64) (drivers []dao.Driver, err error) {
	return dbBase.ListLocalFileDriver(ctx, db.db, deviceId)
}

func (db *DB) GetDriverLocalFile(ctx context.Context, driverId uint64) (driver dao.Driver, err error) {
	return dbBase.GetDriverLocalFile(ctx, db.db, driverId)
}

func (db *DB) GetDriverFileSize(ctx context.Context, driverId uint64) (n uint64, err error) {
	return dbBase.GetDriverFileSize(ctx, db.db, driverId)
}

func (db *DB) GetDriverFileCount(ctx context.Context, driverId uint64) (n uint64, err error) {
	return dbBase.GetDriverFileCount(ctx, db.db, driverId)
}

func (db *DB) GetDriverDirCount(ctx context.Context, driverId uint64) (n uint64, err error) {
	return dbBase.GetDriverDirCount(ctx, db.db, driverId)
}

func (db *DB) GetDriverDirCalculatedInfo(ctx context.Context, driverId uint64, filePath []string) (info dao.DirCalculatedInfo, err error) {
	return dbBase.GetDriverDirCalculatedInfo(ctx, db.db, driverId, filePath)
}

File: ui/packages/mui/api/driver.js
import { noteError } from "@kfs/mui/components/Notification/Notification";
import { getSysConfig } from "@kfs/mui/hox/sysConfig";
import { httpPostBody as httpPostBodyLocal } from "./localServer";
import { httpDelete, httpGet, httpPost } from "./webServer";

export async function listDriver() {
    try {
        console.log('api.listDriver');
        let drivers = await httpGet("/api/v1/drivers");
        return drivers;
    } catch (e) {
        noteError("获取云盘列表失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function newDriver(name, description) {
    try {
        console.log('api.newDriver', name, description);
        let exist = await httpPost("/api/v1/drivers", { name, description });
        if (exist) {
            throw new Error("云盘名称重复: " + name);
        }
    } catch (e) {
        noteError("创建云盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function newDriverBaiduPhoto(name, description, code) {
    try {
        console.log('api.newDriverBaiduPhoto', name, description, code);
        let exist = await httpPost("/api/v1/driverBaiduPhotos", { name, description, code });
        if (exist) {
            throw new Error("云盘名称重复: " + name);
        }
    } catch (e) {
        noteError("创建云盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function newLocalFileDriver(name, description, deviceId, srcPath, encoder, concurrent) {
    try {
        console.log('api.newLocalFileDriver', name, description, srcPath, encoder, concurrent);
        let exist = await httpPost("/api/v1/driverLocalFiles", { name, description, deviceId, srcPath, encoder, concurrent });
        if (exist) {
            throw new Error("云盘名称重复: " + name);
        }
    } catch (e) {
        noteError("创建云盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function resetDriver(driverId) {
    try {
        console.log('api.deleteDriver', driverId);
        await httpGet("/api/v1/drivers/reset", { driverId });
    } catch (e) {
        noteError("重置云盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function deleteDriver(driverId) {
    try {
        console.log('api.deleteDriver', driverId);
        await httpDelete("/api/v1/drivers", { driverId });
    } catch (e) {
        noteError("删除云盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function listLocalFileDriver(deviceId) {
    try {
        console.log('api.listLocalFileDriver', deviceId);
        return await httpGet("/api/v1/listLocalFileDriver", { deviceId });
    } catch (e) {
        noteError("获取本地文件备份盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function startAllLocalFileSync(drivers) {
    try {
        let serverAddr = getSysConfig().socketServer;
        console.log('api.startAllLocalFileSync', serverAddr, drivers);
        return await httpPostBodyLocal("/api/v1/startAllLocalFileSync", { serverAddr, drivers });
    } catch (e) {
        noteError("启动本地文件备份盘失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function getDriverSync(id) {
    try {
        console.log('web.getDriverSync', id);
        return await httpGet("/api/v1/getDriverSync", { id });
    } catch (e) {
        noteError("获取同步配置失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function getDriverLocalFile(driverId) {
    try {
        console.log('web.getDriverLocalFile', driverId);
        return await httpGet("/api/v1/getDriverLocalFile", { driverId });
    } catch (e) {
        noteError("获取本地文件备份配置失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function updateDriverSync(driverId, sync, h, m) {
    try {
        console.log('web.updateDriverSync', driverId, sync, h, m);
        return await httpGet("/api/v1/updateDriverSync", { driverId, sync, h, m });
    } catch (e) {
        noteError("设置定时同步失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function updateDriverLocalFile(driverId, srcPath, ignores, encoder) {
    try {
        console.log('web.updateDriverLocalFile', driverId, srcPath, ignores, encoder);
        return await httpGet("/api/v1/updateDriverLocalFile", { driverId, srcPath, ignores, encoder });
    } catch (e) {
        noteError("设置定时同步失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}

export async function getDriversDirCalculatedInfo(driverId, filePath) {
    try {
        console.log('web.getDriversDirCalculatedInfo', driverId, filePath);
        return await httpGet("/api/v1/drivers/dirCalculatedInfo", { driverId, filePath });
    } catch (e) {
        noteError("获取云盘目录计算属性失败：" + (typeof e.response?.data === 'string' ? e.response?.data : e.message));
        throw e;
    }
}
